File: src/main/java/api/modules/utils/Rotation.java
package api.modules.utils;

import java.io.Serializable;

public class Rotation implements Serializable {

	private static final long serialVersionUID = -5080728242440736168L;
	public Double x;
	public Double y;
	public Double z;
	
	public Rotation () {}

	public Rotation (Rotation position) {
		this.x = position.getX();
		this.y = position.getY();
		this.z = position.getZ();
	}
	
	public Double getX() {
		return x;
	}

	public void setX(Double x) {
		this.x = x;
	}

	public Double getY() {
		return y;
	}

	public void setY(Double y) {
		this.y = y;
	}

	public Double getZ() {
		return z;
	}

	public void setZ(Double z) {
		this.z = z;
	}
	
	public String toString() {
		return "'rotation':{'x':" + getX() + ", 'y':" + getY() + ", 'z':" +getZ() + "}" ;
	}
}

File: prototype/Maze3DAstraSmartyPans/Assets/Assets/Scripts/utils/Rotation.cs
using System;
using UnityEngine;

namespace AssemblyCSharp{

	[Serializable]
	public class Rotation {

		public float x;
		public float y;
		public float z;

		public Rotation (){}

		public Rotation (Transform transform) {
			this.x = transform.rotation.x;
			this.y = transform.rotation.y;
			this.z = transform.rotation.z;
		}
	}
}

